### PR TITLE
Add support for persisting plan in ZooKeeper for a tenant

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/notification/NotificationsDbTest.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import com.yahoo.vespa.hosted.controller.tenant.LastLoginInfo;
 import com.yahoo.vespa.hosted.controller.tenant.TenantContacts;
 import com.yahoo.vespa.hosted.controller.tenant.TenantInfo;
+import org.apache.zookeeper.Op;
```
Comment (freva) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/notification/NotificationsDbTest.java:30]:
```suggestion
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/TenantSerializer.java [hunk 2]
```diff
@@ -215,7 +217,10 @@ private CloudTenant cloudTenantFrom(Inspector tenantObject) {
         Instant tenantRolesLastMaintained = SlimeUtils.instant(tenantObject.field(tenantRolesLastMaintainedField));
         List<CloudAccountInfo> cloudAccountInfos = cloudAccountsFromSlime(tenantObject.field(cloudAccountsField));
         Optional<BillingReference> billingReference = billingReferenceFrom(tenantObject.field(billingReferenceField));
-        return new CloudTenant(name, createdAt, lastLoginInfo, creator, developerKeys, info, tenantSecretStores, archiveAccess, invalidateUserSessionsBefore, tenantRolesLastMaintained, cloudAccountInfos, billingReference);
+        Optional<String> planId = planId(tenantObject.field(planIdField));
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/TenantSerializer.java:220]: Consider simplifying with something like
```suggestion
        Optional<String> planId = SlimeUtils.optionalString(tenantObject.field(planIdField));
```

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/tenant/CloudTenant.java [hunk 3]
```diff
@@ -28,13 +28,15 @@ public class CloudTenant extends Tenant {
     private final ArchiveAccess archiveAccess;
     private final Optional<Instant> invalidateUserSessionsBefore;
     private final Optional<BillingReference> billingReference;
+    private final Optional<String> planId;
```
Comment (freva) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/tenant/CloudTenant.java:31]: Use `PlanId` type?

**Review by freva — DISMISSED** [review]
Maybe we should default to `none` type in ZK deserialization code so we can avoid the `Optional`? The future maintainer will update to the actual plan before we start using this anyway.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/tenant/CloudTenant.java (hunk 3 above)
Comment (hmusum) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/tenant/CloudTenant.java]: Yep, done

**Review by freva — APPROVED** [review]

**Pull request merged**